sql/generator.go

```diff
@@ -0,0 +1,57 @@
+package sql
+
+import (
+	"io"
+
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+// Generator will generate a set of values for a given row.
+type Generator interface {
+	// Next value in the generator.
+	Next() (interface{}, error)
+	// Close the generator and dispose resources.
+	Close() error
+}
+
+// ErrNotGenerator is returned when the value cannot be converted to a
+// generator.
+var ErrNotGenerator = errors.NewKind("cannot convert value of type %T to a generator")
+
+// ToGenerator converts a value to a generator if possible.
+func ToGenerator(v interface{}) (Generator, error) {
```

Comment:
Since it only really converts arrays, could be called ArrayToGenerator.

Reply:
ArrayToGenerator suggests you need to pass an array and get a generator. This, however, receives any kind of value and returns a Generator or an error if the value could not be converted. The fact that arrays are converted is an implementation detail.

Reply:
It is expected to convert other types in the future? (like maps for example).

Reply:
Nope, because there are not any other types that can be a generator besides  arrays and `Generator`. But still calling it ArrayToGenerator implies the input is always an Array, which is not the case. It's meant to return the generator as is if the input is a generator, convert array to a generator and return it or fail in any other case.

Reply:
Mmm, then since the `interface{}` doesn't provide documentation about it to the user this could be specified in the function doc.

Reply:
The user does not need to know which types are handled here. Only needs to know that if the value can be converted to a generator it will, if not, it will error.